README.md
# wgrib
WGRIB is a program to manipulate, inventory and decode [GRIB](https://en.wikipedia.org/wiki/GRIB) (edition 1) files.

**This is not the official version**

Forked from the official version available at [NCEP](https://www.cpc.ncep.noaa.gov/products/wesley/wgrib.html)


## Usage

### Short inventory
```
> wgrib.exe waves.grb
1:0:d=21011800:SWH:kpds5=229:kpds6=102:kpds7=0:TR=10:P1=0:P2=168:TimeU=1:MSL:168hr fcst:type=Perturbed forecast 30:NAve=0
2:396:d=21011800:SWH:kpds5=229:kpds6=102:kpds7=0:TR=10:P1=0:P2=168:TimeU=1:MSL:168hr fcst:type=Perturbed forecast 34:NAve=0
3:792:d=21011800:SWH:kpds5=229:kpds6=102:kpds7=0:TR=10:P1=0:P2=168:TimeU=1:MSL:168hr fcst:type=Perturbed forecast 47:NAve=0
4:1188:d=21011800:SWH:kpds5=229:kpds6=102:kpds7=0:TR=10:P1=0:P2=168:TimeU=1:MSL:168hr fcst:type=Perturbed forecast 11:NAve=0
5:1584:d=21011800:SWH:kpds5=229:kpds6=102:kpds7=0:TR=10:P1=0:P2=168:TimeU=1:MSL:168hr fcst:type=Perturbed forecast 9:NAve=0
6:1980:d=21011800:SWH:kpds5=229:kpds6=102:kpds7=0:TR=10:P1=0:P2=168:TimeU=1:MSL:168hr fcst:type=Perturbed forecast 49:NAve=0
7:2376:d=21011800:SWH:kpds5=229:kpds6=102:kpds7=0:TR=10:P1=0:P2=168:TimeU=1:MSL:168hr fcst:type=Perturbed forecast 31:NAve=0
8:2772:d=21011800:SWH:kpds5=229:kpds6=102:kpds7=0:TR=10:P1=0:P2=168:TimeU=1:MSL:168hr fcst:type=Perturbed forecast 41:NAve=0
9:3168:d=21011800:SWH:kpds5=229:kpds6=102:kpds7=0:TR=10:P1=0:P2=168:TimeU=1:MSL:168hr fcst:type=Perturbed forecast 50:NAve=0
```


## Changes compared to the official version (2021-01-22)
* Perturbed forecast numbering for additional ECMWF streams


src/Changes
v1.8.1.0b 5/10 PDStimes .. decided to remove normalize minutes to hours, make time units consistent in inv
v1.8.1.0  5/10 PDStimes .. if time_range == 10 and time_unit == 0 (minute) .. normalize to hour only
		occurs when (time/256 % 60) == 0 && ((time % 256) % 60) == 0 rather than (time % 60 == 0)
v1.8.0.13g 5/10 changed decl of pos so error message wont appear on AIX
v1.8.0.13f 3/10 added code from Brian Doty for large ECMWF files
v1.8.0.13e 2/10 added level 216/217/239/241
		updated nceptab 128, 129
           9/09 warning message if grib2 message encountered
v1.8.0.13d 9/09 added ncep table 141, fixed typo in ncep table 129 (thanks YT)
v1.8.0.13c 6/09 Dusan Jovic's changes for Arakawa staggered b-grid (205)
v1.8.0.13b 6/09 updated ncep table 129
v1.8.0.13a 5/09 added SUNSD ncep table 133
v1.8.0.13 5/09 added -cmc by NCO request, updated ncep table 129
v1.8.0.12z: 4/09 updated/added ectables: 128, 129, 131, 132, 133, 140, 150, 
    151, 160, 162, 170, 171, 173, 174, 180, 190, 200, 210, 211, 228
v1.8.0.12y: 3/09 added #include <stdlib.h> to BDSunpk.c (Lulin Song)
	fix: bit 1 table 7 doesn't apply to lambert conformal
v1.8.0.12x: 11/08 make center LAMI point to DWD tables (Davide Sacchetti)  (removed)
v1.8.0.12w: 9/08 (preliminary) MDL table
v1.8.0.12v: 8/08 polar & lambert conformal grid: dx/dy now respect bit 1 of resolution flag
v1.8.0.12u: 4/08 update time ranges for CFSRR
v1.8.0.12t: 3/08 update table ncep table 129, fix ncep table 133, #205
v1.8.0.12s: 1/08 update ncep table 133 (CFSRR)
v1.8.0.12r: 12/07 update ncep table 133 (CFSRR)
v1.8.0.12q: 11/07 update ecmwf table 128
v1.8.0.12p: 10/07 update ncep table 129, 130, add ncep table 140
v1.8.0.12o: 5/07 update ncep table 129
v1.8.0.12n: 3/07 name change DH->VEDH (ncep table 129), update ncep table 129, 130
                 updated levels table to current on388
                 updated levels table to current on388
v1.8.0.12m: 3/07 added FAST_GRIBTAB compile option, support for updates in user defined grib tables
v1.8.0.12l: 3/07 minor table change ncep 129 (lat/lon of u, v and p points)
v1.8.0.12k: 2/07 stdin 2000 characters now
v1.8.0.12j: 2/07 level 210 corresponds to WMO standard for non-NCEP files
v1.8.0.12i: 12/06 ncep table 128 entry 186  K->C
v1.8.0.12h: 9/06 can figure out the nxny of complex packed files, still no support for decoding
	     so you can manipulate ec-spectral files but can't decode them
v1.8.0.12g: 8/06 more ncep ensemble information 
v1.8.0.12f: 8/06 more ncep ensemble information stddev/wtd mean
v1.8.0.12e: 8/06 level 20: T=280K, added verbose parameter, revised level 113
v1.8.0.12d: 8/06 added marine branch table 128 (extremely revised)
v1.8.0.12c: 8/06 level 20
v1.8.0.12b: 7/06 update ncep tab 129, remove ncep tab 128 .. need new table, minor fixes
v1.8.0.12a: 6/06 replace error message - complex packing spectral files
v1.8.0.12: 5/06 update thinned gaussian grid info for grib2ctl.pl
v1.8.0.11: 5/06 flag bad read_grib calls
v1.8.0.10c: 1/06 fix the 1/3/05 additions to PDStimes,  H. Frank
v1.8.0.10b: 1/06 add new (204, 205) and updated DWD tables  H. Frank, 10-19-2005
v1.8.0.10a: 12/05 add ectable 172 from Geert Jan van Oldenborgh (KNMI)
v1.8.0.10: 9/05 fix gribtable.c 
v1.8.0.9m1: 9/05 put copyright info in wgrib_main.c and wgrib.c
v1.8.0.9m: 8/05 add time range 6 and 7, updated add_time code
v1.8.0.9l: 5/05 update ncep 129 table, new levels done by D. Stokes
v1.8.0.9k: 4/05 ec table 190, more ec diagnostics (ec_ext)
v1.8.0.9j: 4/05 minor comment changes in ncep table 131 (RR)
v1.8.0.9i: 12/04 fixed typo introduced 11/04 (table 129, entry 192)
v1.8.0.9h: 11/04 updated table ncep-129 (strange entry)
v1.8.0.9g: 9/04 fix if (gds && GDS_Harmonic_type(gds) == 1)
           found by Graziano Giuliani 
v1.8.0.9f: 8/04 added ecmwf table 151
v1.8.0.9e: 8/04 changes of CFS time ranges
v1.8.0.9d: 7/04 fixed reanal grib table NLON -> ELON
v1.8.0.9c: 6/04 chaged UVI defn ncep opn table
v1.8.0.9b: 6/04 added table 132 (global reanalysis)
v1.8.0.9a: 6/04 updated table 129,130
v1.8.0.9: 5/04 added new time codes 132-135, changed time code 51 (clim)
v1.8.0.8: 4/04 added -ncep_ens option
v1.8.0.7e: 3/04 multiple user tables (fix open)
v1.8.0.7d: 3/04 extern minute -> extern int minute
v1.8.0.7c: 3/04 for clim (51) don't print 'anl'
v1.8.0.7b: 2/04 update NCEP grib table 129
v1.8.0.7a: 11/03 update NCEP grib table 131
v1.8.0.7: 10/03 Mods fron Norwegian Meteorological Institute
                fixed ec table 131, and enhansed ex_ext.c
v1.8.0.6: 9/03 added time codes 128-131 (for RR)
v1.8.0.5: 9/03 fix scan mode and change format
v1.8.0.4: 9/03 new dwd tables and -dwdgrib option (thanks to Helmut P. Frank)
               updated formats_update.txt for -dwdgrib option
v1.8.0.3k: 7/03 fixed N/S vs grid wind diagnostic with -v option
                  found by Dusan Jovic
v1.8.0.3j: 6/03 fixed level 206 (was 205)
v1.8.0.3i: 5/03 updated ncep table 131 
v1.8.0.3h: 5/03 updated ncep table 131
v1.8.0.3g: 4/03 ecmwf table 128, 140 (names from ECMWF web site), updated 131
v1.8.0.3f: 4/03 updated ncep131, ncep130, ncep-opn
v1.8.0.3e: 3/03 fix level 141, update tables 130 and 131
v1.8.0.3d: 2/03 added prelim ncep grib tables 130 and 131
v1.8.0.3c: 1/03 Reanalysis Project updates: scan mode in english, 
    grid vs NS winds, and dusan.jovic@noaa.gov: ss2dEGRID support
v1.8.0.3b: 1/03 minor update: changed units of CICE in NCEP tables
v1.8.0.3a: 11/02 update ncep_opn and add ncep_129 table
v1.8.0.3: 10/02 update levels and ncep_opn grib table
v1.8.0.2: 10/02 check gds before grib_check, redid level 117
v1.8.0.1: 10/02 added cptec gribtable 254
v1.8: 4/1/02: Decode of simple packed spectral data and some cleanup of the
     triangular grid (Luis Kornblueh, Max-Plank Institute of Meteorology)
v1.7.4.1a 12/01: added new entry to OMB grib table (REV)
v1.7.4.1 12/01: D. Haalman (METEO SERVICE weather research GmbH) 
               rewrote seekgrib.c
v1.7.4   10/01: Merged changes made by  Helmut P. Frank (DWD) to current 

src/nceptab_129.c
#include "cnames.h"

const struct ParmTable parm_table_nceptab_129[256] = {
      /* 0 */ {"var0", "undefined"},
      /* 1 */ {"PRES", "Pressure [Pa]"},
      /* 2 */ {"PRMSL", "Pressure reduced to MSL [Pa]"},
      /* 3 */ {"PTEND", "Pressure tendency [Pa/s]"},
      /* 4 */ {"PVORT", "Pot. vorticity [km^2/kg/s]"},
      /* 5 */ {"ICAHT", "ICAO Standard Atmosphere Reference Height [M]"},
      /* 6 */ {"GP", "Geopotential [m^2/s^2]"},
      /* 7 */ {"HGT", "Geopotential height [gpm]"},
      /* 8 */ {"DIST", "Geometric height [m]"},
      /* 9 */ {"HSTDV", "Std dev of height [m]"},
      /* 10 */ {"TOZNE", "Total ozone [Dobson]"},
      /* 11 */ {"TMP", "Temp. [K]"},
      /* 12 */ {"VTMP", "Virtual temp. [K]"},
      /* 13 */ {"POT", "Potential temp. [K]"},
      /* 14 */ {"EPOT", "Pseudo-adiabatic pot. temp. [K]"},
      /* 15 */ {"TMAX", "Max. temp. [K]"},
      /* 16 */ {"TMIN", "Min. temp. [K]"},
      /* 17 */ {"DPT", "Dew point temp. [K]"},
      /* 18 */ {"DEPR", "Dew point depression [K]"},
      /* 19 */ {"LAPR", "Lapse rate [K/m]"},
      /* 20 */ {"VIS", "Visibility [m]"},
      /* 21 */ {"RDSP1", "Radar spectra (1) [non-dim]"},
      /* 22 */ {"RDSP2", "Radar spectra (2) [non-dim]"},
      /* 23 */ {"RDSP3", "Radar spectra (3) [non-dim]"},
      /* 24 */ {"PLI", "Parcel lifted index (to 500 hPa) [K]"},
      /* 25 */ {"TMPA", "Temp. anomaly [K]"},
      /* 26 */ {"PRESA", "Pressure anomaly [Pa]"},
      /* 27 */ {"GPA", "Geopotential height anomaly [gpm]"},
      /* 28 */ {"WVSP1", "Wave spectra (1) [non-dim]"},
      /* 29 */ {"WVSP2", "Wave spectra (2) [non-dim]"},
      /* 30 */ {"WVSP3", "Wave spectra (3) [non-dim]"},
      /* 31 */ {"WDIR", "Wind direction [deg]"},
      /* 32 */ {"WIND", "Wind speed [m/s]"},
      /* 33 */ {"UGRD", "u wind [m/s]"},
      /* 34 */ {"VGRD", "v wind [m/s]"},
      /* 35 */ {"STRM", "Stream function [m^2/s]"},
      /* 36 */ {"VPOT", "Velocity potential [m^2/s]"},
      /* 37 */ {"MNTSF", "Montgomery stream function [m^2/s^2]"},
      /* 38 */ {"SGCVV", "Sigma coord. vertical velocity [/s]"},
      /* 39 */ {"VVEL", "Pressure vertical velocity [Pa/s]"},
      /* 40 */ {"DZDT", "Geometric vertical velocity [m/s]"},
      /* 41 */ {"ABSV", "Absolute vorticity [/s]"},
      /* 42 */ {"ABSD", "Absolute divergence [/s]"},
      /* 43 */ {"RELV", "Relative vorticity [/s]"},
      /* 44 */ {"RELD", "Relative divergence [/s]"},
      /* 45 */ {"VUCSH", "Vertical u shear [/s]"},
      /* 46 */ {"VVCSH", "Vertical v shear [/s]"},
      /* 47 */ {"DIRC", "Direction of current [deg]"},
      /* 48 */ {"SPC", "Speed of current [m/s]"},
      /* 49 */ {"UOGRD", "u of current [m/s]"},
      /* 50 */ {"VOGRD", "v of current [m/s]"},
      /* 51 */ {"SPFH", "Specific humidity [kg/kg]"},
      /* 52 */ {"RH", "Relative humidity [%]"},
      /* 53 */ {"MIXR", "Humidity mixing ratio [kg/kg]"},
      /* 54 */ {"PWAT", "Precipitable water [kg/m^2]"},
      /* 55 */ {"VAPP", "Vapor pressure [Pa]"},
      /* 56 */ {"SATD", "Saturation deficit [Pa]"},
      /* 57 */ {"EVP", "Evaporation [kg/m^2]"},
      /* 58 */ {"CICE", "Cloud Ice [kg/m^2]"},
      /* 59 */ {"PRATE", "Precipitation rate [kg/m^2/s]"},
      /* 60 */ {"TSTM", "Thunderstorm probability [%]"},
      /* 61 */ {"APCP", "Total precipitation [kg/m^2]"},
      /* 62 */ {"NCPCP", "Large scale precipitation [kg/m^2]"},
      /* 63 */ {"ACPCP", "Convective precipitation [kg/m^2]"},
      /* 64 */ {"SRWEQ", "Snowfall rate water equiv. [kg/m^2/s]"},
      /* 65 */ {"WEASD", "Accum. snow [kg/m^2]"},
      /* 66 */ {"SNOD", "Snow depth [m]"},
      /* 67 */ {"MIXHT", "Mixed layer depth [m]"},
      /* 68 */ {"TTHDP", "Transient thermocline depth [m]"},
      /* 69 */ {"MTHD", "Main thermocline depth [m]"},
      /* 70 */ {"MTHA", "Main thermocline anomaly [m]"},
      /* 71 */ {"TCDC", "Total cloud cover [%]"},
      /* 72 */ {"CDCON", "Convective cloud cover [%]"},
      /* 73 */ {"LCDC", "Low level cloud cover [%]"},
      /* 74 */ {"MCDC", "Mid level cloud cover [%]"},
      /* 75 */ {"HCDC", "High level cloud cover [%]"},
      /* 76 */ {"CWAT", "Cloud water [kg/m^2]"},
      /* 77 */ {"BLI", "Best lifted index (to 500 hPa) [K]"},
      /* 78 */ {"SNOC", "Convective snow [kg/m^2]"},
      /* 79 */ {"SNOL", "Large scale snow [kg/m^2]"},
      /* 80 */ {"WTMP", "Water temp. [K]"},
      /* 81 */ {"LAND", "Land cover (land=1;sea=0) [fraction]"},
      /* 82 */ {"DSLM", "Deviation of sea level from mean [m]"},
      /* 83 */ {"SFCR", "Surface roughness [m]"},
      /* 84 */ {"ALBDO", "Albedo [%]"},
      /* 85 */ {"TSOIL", "Soil temp. [K]"},
      /* 86 */ {"SOILM", "Soil moisture content [kg/m^2]"},
      /* 87 */ {"VEG", "Vegetation [%]"},
      /* 88 */ {"SALTY", "Salinity [kg/kg]"},
      /* 89 */ {"DEN", "Density [kg/m^3]"},
      /* 90 */ {"WATR", "Water runoff [kg/m^2]"},
      /* 91 */ {"ICEC", "Ice concentration (ice=1;no ice=0) [fraction]"},
      /* 92 */ {"ICETK", "Ice thickness [m]"},
      /* 93 */ {"DICED", "Direction of ice drift [deg]"},
      /* 94 */ {"SICED", "Speed of ice drift [m/s]"},
      /* 95 */ {"UICE", "u of ice drift [m/s]"},
      /* 96 */ {"VICE", "v of ice drift [m/s]"},
      /* 97 */ {"ICEG", "Ice growth rate [m/s]"},
      /* 98 */ {"ICED", "Ice divergence [/s]"},
      /* 99 */ {"SNOM", "Snow melt [kg/m^2]"},
      /* 100 */ {"HTSGW", "Sig height of wind waves and swell [m]"},
      /* 101 */ {"WVDIR", "Direction of wind waves [deg]"},
      /* 102 */ {"WVHGT", "Sig height of wind waves [m]"},
      /* 103 */ {"WVPER", "Mean period of wind waves [s]"},
      /* 104 */ {"SWDIR", "Direction of swell waves [deg]"},
      /* 105 */ {"SWELL", "Sig height of swell waves [m]"},
      /* 106 */ {"SWPER", "Mean period of swell waves [s]"},
      /* 107 */ {"DIRPW", "Primary wave direction [deg]"},
      /* 108 */ {"PERPW", "Primary wave mean period [s]"},
      /* 109 */ {"DIRSW", "Secondary wave direction [deg]"},
      /* 110 */ {"PERSW", "Secondary wave mean period [s]"},
      /* 111 */ {"NSWRS", "Net short wave (surface) [W/m^2]"},
      /* 112 */ {"NLWRS", "Net long wave (surface) [W/m^2]"},
      /* 113 */ {"NSWRT", "Net short wave (top) [W/m^2]"},
      /* 114 */ {"NLWRT", "Net long wave (top) [W/m^2]"},
      /* 115 */ {"LWAVR", "Long wave [W/m^2]"},
      /* 116 */ {"SWAVR", "Short wave [W/m^2]"},
      /* 117 */ {"GRAD", "Global radiation [W/m^2]"},
      /* 118 */ {"BRTMP", "Brightness temperature [K]"},
      /* 119 */ {"LWRAD", "Radiance with respect to wave no. [W/m/sr]"},
      /* 120 */ {"SWRAD", "Radiance with respect ot wave len. [W/m^3/sr]"},
      /* 121 */ {"LHTFL", "Latent heat flux [W/m^2]"},
      /* 122 */ {"SHTFL", "Sensible heat flux [W/m^2]"},
      /* 123 */ {"BLYDP", "Boundary layer dissipation [W/m^2]"},
      /* 124 */ {"UFLX", "Zonal momentum flux [N/m^2]"},
      /* 125 */ {"VFLX", "Meridional momentum flux [N/m^2]"},
      /* 126 */ {"WMIXE", "Wind mixing energy [J]"},
      /* 127 */ {"IMGD", "Image data []"},
      /* 128 */ {"PAOT", "Probability anomaly of temp [%]"},
      /* 129 */ {"PAOP", "Probability anomaly of precip [%]"},
      /* 130 */ {"CWR", "Probability of wetting rain > 0.1 in [%]"},
      /* 131 */ {"FRAIN", "Rain fraction of total liquid water []"},
      /* 132 */ {"FICE", "Ice fraction of total condensate []"},
      /* 133 */ {"FRIME", "Rime factor []"},
      /* 134 */ {"CUEFI", "Convective cloud efficiency []"},
      /* 135 */ {"TCOND", "Total condensate [kg/kg]"},
      /* 136 */ {"TCOLW", "Total column cloud water [kg/m/m]"},
      /* 137 */ {"TCOLI", "Total column cloud ice [kg/m/m]"},
      /* 138 */ {"TCOLR", "Total column rain [kg/m/m]"},
      /* 139 */ {"TCOLS", "Total column snow [kg/m/m]"},
      /* 140 */ {"TCOLC", "Total column condensate [kg/m/m]"},
      /* 141 */ {"PLPL", "Pressure of level from which parcel was lifted [Pa]"},
      /* 142 */ {"HLPL", "Height of level from which parcel was lifted [m]"},
      /* 143 */ {"CEMS", "Cloud Emissivity [fraction]"},
      /* 144 */ {"COPD", "Cloud Optical Depth [non-dim]"},
      /* 145 */ {"PSIZ", "Effective Particle size [microns]"},
      /* 146 */ {"TCWAT", "Total Water Cloud [%]"},
      /* 147 */ {"TCICE", "Total Ice Cloud [%]"},
      /* 148 */ {"WDIF", "Wind Difference [m/s]"},
      /* 149 */ {"WSTP", "Wave Steepness [non-dim]"},
      /* 150 */ {"PTAN", "Probability of Temp. above normal [%]"},
      /* 151 */ {"PTNN", "Probability of Temp. near normal [%]"},
      /* 152 */ {"PTBN", "Probability of Temp. below normal [%]"},
      /* 153 */ {"PPAN", "Probability of Precip. above normal [%]"},
      /* 154 */ {"PPNN", "Probability of Precip. near normal [%]"},
      /* 155 */ {"PPBN", "Probability of Precip. below normal [%]"},
      /* 156 */ {"PMTC", "Particulate matter (coarse) [ug/m^3]"},
      /* 157 */ {"PMTF", "Particulate matter (fine) [ug/m^3]"},
      /* 158 */ {"AETMP", "Analysis Error of Temperature [K]"},
      /* 159 */ {"AEDPT", "Analysis Error of Dew Point [K]"},
      /* 160 */ {"AESPH", "Analysis Error of Specific Humidity [kg/kg] wne"},
      /* 161 */ {"AEUWD", "Analysis Error of U-wind [m/s]"},
      /* 162 */ {"AEVWD", "Analysis Error of V-wind [m/s]"},
      /* 163 */ {"LPMTF", "Particulate matter (fine) [log10(ug/m^3)]"},
      /* 164 */ {"LIPMF", "Integrated Column Particulate matter (fine) [log10(ug/m^2)] wne"},
      /* 165 */ {"REFZR", "Derived radar reflectivity backscatter from rain [mm^6/m^3]"},
      /* 166 */ {"REFZI", "Derived radar reflectivity backscatter from ice [mm^6/m^3]"},
      /* 167 */ {"REFZC", "Derived radar reflectivity backscatter from parameterized convection [mm^6/m^3]"},
      /* 168 */ {"TCLSW", "Integrated supercooled liquid water [kg/m^2]"},
      /* 169 */ {"TCOLM", "Total Column Integrated Melting Ice [kg/m^2]"},
      /* 170 */ {"ELRDI", "Ellrod Index [non-dim]"},
      /* 171 */ {"TSEC", "Seconds prior to initial reference time [sec]"},
      /* 172 */ {"TSECA", "Seconds after initial reference time [sec]"},
      /* 173 */ {"NUM", "Number of samples/observations [non-dim]"},
      /* 174 */ {"AEPRS", "Analysis Error of Pressure [Pa]"},
      /* 175 */ {"ICSEV", "Icing Severity [non-dim]"},
      /* 176 */ {"ICPRB", "Icing Probability [non-dim]"},
      /* 177 */ {"LAVNI", "Low-level Aviation Interest [non-dim]"},
      /* 178 */ {"HAVNI", "High-level Aviation Interest [non-dim]"},
      /* 179 */ {"FLGHT", "Flight Category [non-dim]"},
      /* 180 */ {"OZCON", "Ozone concentration [ppb]"},
      /* 181 */ {"OZCAT", "Categorical ozone concentration [?]"},
      /* 182 */ {"VEDH", "vertical heat eddy diffusivity [m^2/s]"},
      /* 183 */ {"SIGV", "Sigma level value [non-dim]"},
      /* 184 */ {"EWGT", "Ensemble Weight [non-dim]"},
      /* 185 */ {"CICEL", "Confidence indicator - Ceiling [non-dim]"},
      /* 186 */ {"CIVIS", "Confidence indicator - Visibility [non-dim] 186"},
      /* 187 */ {"var187", "undefined"},
      /* 188 */ {"LAVV", "Latitude of V wind component of velocity [deg]"},
      /* 189 */ {"LOVV", "Longitude of V wind component of velocity [deg]"},
      /* 190 */ {"USCT", "Scatterometer est. U wind component [m/s]"},
      /* 191 */ {"VSCT", "Scatterometer est. V wind component [m/s]"},
      /* 192 */ {"LAUV", "Latitude of U wind component of velocity [deg]"},
      /* 193 */ {"LOUV", "Longitude of U wind component of velocity [deg]"},
      /* 194 */ {"TCHP", "Tropical Cyclone Heat Potential [J/m^2]"},
      /* 195 */ {"DBSS", "Geometric Depth Below Sea Surface [m]"},
      /* 196 */ {"ODHA", "Ocean Dynamic Heat Anomaly [dynamic m]"},
      /* 197 */ {"OHC", "Ocean Heat Content [J/m^2]"},
      /* 198 */ {"SSHG", "Sea Surface Height Relative to Geoid [m]"},
      /* 199 */ {"SLTFL", "Salt flux [g/cm^2/s]"},
      /* 200 */ {"DUVB", "UV-B Downward Solar Flux [W/m^2]"},
      /* 201 */ {"CDUVB", "Clear Sky UV-B Downward Solar Flux [W/m^2]"},
      /* 202 */ {"THFLX", "Total downward heat flux at surface [W/m^2]"},
      /* 203 */ {"UVAR", "U velocity variance [m^2/s^2]"},
      /* 204 */ {"VVAR", "V velocity variance [m^2/s^2]"},
      /* 205 */ {"UVVCC", "UV Velocity Cross Correlation [m^2/s^2]"},
      /* 206 */ {"MCLS", "Meteorological Correlation Length Scale [m]"},
      /* 207 */ {"LAPP", "Latitude of pressure point [deg]"},
      /* 208 */ {"LOPP", "Longitude of pressure point [deg]"},
      /* 209 */ {"var209", "undefined"},
      /* 210 */ {"REFO", "Observed radar reflectivity [dbZ]"},
      /* 211 */ {"REFD", "Derived radar reflectivity [dbZ]"},
      /* 212 */ {"REFC", "Maximum/Composite radar reflectivity [dbZ]"},
      /* 213 */ {"SBT122", "Simulated Brightness Temperature for GOES12, Channel 2 [K]"},
      /* 214 */ {"SBT123", "Simulated Brightness Temperature for GOES12, Channel 3 [K]"},
      /* 215 */ {"SBT124", "Simulated Brightness Temperature for GOES12, Channel 4 [K]"},
      /* 216 */ {"SBT125", "Simulated Brightness Temperature for GOES12, Channel 5 [K]"},
      /* 217 */ {"MINRH", "Minimum Relative Humumidity [%]"},
      /* 218 */ {"MAXRH", "Maximum Relative Humumidity [%]"},
      /* 219 */ {"CEIL", "Ceiling [m]"},
      /* 220 */ {"PBLREG", "Planetary boundary layer regime []"},
      /* 221 */ {"SBC123", "Simulated brightness counts for GOES12, Channel 3 [byte]"},
      /* 222 */ {"SBC124", "Simulated brightness counts for GOES12, Channel 4 [byte]"},
      /* 223 */ {"RPRATE", "Rain precipitation rate [kg/m^2/s]"},
      /* 224 */ {"SPRATE", "Snow precipitation rate [kg/m^2/s]"},
      /* 225 */ {"FPRATE", "Freezing rain precipitation rate [kg/m^2/s]"},
      /* 226 */ {"IPRATE", "Ice pellets precipitation rate [kg/m^2/s]"},
      /* 227 */ {"UPHL", "Updraft Helicity [m^2/s^2]"},
      /* 228 */ {"SURGE", "Storm Surge [m]"},
      /* 229 */ {"ETSRG", "Extra-tropical storm Surge [m]"},
      /* 230 */ {"RHPW", "Relative humidity with respect to precip water [%]"},
      /* 231 */ {"OZMAX1", "Ozone daily max from 1-hour ave [ppbV]"},
      /* 232 */ {"OZMAX8", "Ozone daily max from 8-hour ave [ppbV]"},
      /* 233 */ {"var233", "undefined"},
      /* 234 */ {"var234", "undefined"},
      /* 235 */ {"var235", "undefined"},
      /* 236 */ {"var236", "undefined"},
      /* 237 */ {"var237", "undefined"},
      /* 238 */ {"var238", "undefined"},
      /* 239 */ {"var239", "undefined"},
      /* 240 */ {"var240", "undefined"},
      /* 241 */ {"var241", "undefined"},
      /* 242 */ {"TCSRG20", "20% tropical cyclone storm exceedance [m]"},
      /* 243 */ {"TCSRG30", "30% tropical cyclone storm exceedance [m]"},
      /* 244 */ {"TCSRG40", "40% tropical cyclone storm exceedance [m]"},
      /* 245 */ {"TCSRG50", "50% tropical cyclone storm exceedance [m]"},
      /* 246 */ {"TCSRG60", "60% tropical cyclone storm exceedance [m]"},
      /* 247 */ {"TCSRG70", "70% tropical cyclone storm exceedance [m]"},
      /* 248 */ {"TCSRG80", "80% tropical cyclone storm exceedance [m]"},
      /* 249 */ {"TCSRG90", "90% tropical cyclone storm exceedance [m]"},
      /* 250 */ {"RETOP", "Radar echo top (18.3 DBZ) [m]"},
      /* 251 */ {"TENV", "Total energy norm variance []"},
      /* 252 */ {"var252", "undefined"},
      /* 253 */ {"var253", "undefined"},
      /* 254 */ {"var254", "undefined"},
      /* 255 */ {"var255", "undefined"},
};
